File: C++언어/_2020_07_02/Human/Human.cpp
#include "Human.h"

Human::Human(string name, int age)		// default 매개변수, 아무것도 값을 안받아오면 (사람,0) // 받아오면 (그 값)
{
	this->name = name;
	this->age = age;
	cout << "Human 생성자 호출" << endl;
}

Human::~Human()
{
	cout << "Human 소멸자 호출" << endl;
}

void Human::eat()
{
	cout << this->name << "은 밥을 먹는다" << endl;
}
void Human::sleep()
{
	cout << this->name << "은 잠을 잔다" << endl;
}
void Human::work()
{
	cout << this->name<< "은 일을 한다" << endl;
}
void Human::info()
{
	cout << "이름 : " << this->name << endl;
	cout << "나이 : " << this->age << endl;
}

File: C#언어/C# 수업/_2020_07_29/_2020_07_29_Forms/_40_Serialize/Human.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _40_Serialize
{
    [Serializable]  // 이 객체(클래스)는 시리얼라이즈로 읽어들이는게 가능하다.
    class Human
    {
        public string Name { get; set; }
        public int Age { get; set; }
        public float Height { get; set; }
        public float Weight { get; set; }

        public override string ToString()       // 오브젝트 클래스로부터 상속받은 ToString()
        {
            return Name + ":" + Age + ":" + Height + ":" + Weight;
        }
    }
}
